tools/modelgen/generators/templates/register.tmpl

```diff
@@ -0,0 +1,5 @@
+{{ range .Objects }}
+func Register{{ initialCap .Name }}Callbacks(handler {{ initialCap .Name }}Callbacks) {
+  objCallbackHandler.{{ initialCap .Name }}Cb = handler
```

Comment:
Its done based on callback handler being nil for an object in REST handlers..

// Check if we handle this object
if objCallbackHandler.{{initialCap .}}Cb == nil {
log.Errorf("No callback registered for {{.}} object”)
return errors.New("Invalid object type")
}

-sukhesh


From:  Madhav Puri <notifications@github.com<mailto:notifications@github.com>>
Reply-To:  contiv/objmodel <reply@reply.github.com<mailto:reply@reply.github.com>>
Date:  Wednesday, August 19, 2015 at 9:35 AM
To:  contiv/objmodel <objmodel@noreply.github.com<mailto:objmodel@noreply.github.com>>
Subject:  Re: [objmodel] Modelgen refactors (cont.) (#2)


In tools/modelgen/generators/templates/register.tmpl <https://github.com/contiv/objmodel/pull/2#discussion_r37436267>:
@@ -0,0 +1,5 @@
+{{ range .Objects }}
+func Register{{ initialCap .Name }}Callbacks(handler {{ initialCap .Name }}Callbacks) {
+  objCallbackHandler.{{ initialCap .Name }}Cb = handler
May be we should add a check for handler being nil and initialize the
<>Cb to a default no-op handler in that case?
This will permit objmodel clients to just provide callback handlers just for objects they are interested in and rest of handlers will be set to default no-op handler.
—
Reply to this email directly or
view it on GitHub <https://github.com/contiv/objmodel/pull/2/files#r37436267>.